---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    fig.path = "man/figures/README-",
    out.width = "100%"
)
```

# TREG

<!-- badges: start -->
[![Lifecycle: experimental](https://img.shields.io/badge/lifecycle-experimental-orange.svg)](https://lifecycle.r-lib.org/articles/stages.html#experimental)
[![BioC status](http://www.bioconductor.org/shields/build/release/bioc/TREG.svg)](https://bioconductor.org/checkResults/release/bioc-LATEST/TREG)
[![R-CMD-check-bioc](https://github.com/LieberInstitute/TREG/workflows/R-CMD-check-bioc/badge.svg)](https://github.com/LieberInstitute/TREG/actions)
<!-- badges: end -->

The goal of `TREG` is to help find find candidate **Total RNA Expression Genes (TREGs)**
in single nucleus RNA-seq data.

### Why are TREGs useful?
The expression of a TREG is proportional to the the overall RNA expression in a
cell. This relationship can be used to estimate total RNA content in cells in 
assays where only a few genes can be measured, such as single-molecule 
fluorescent in situ hybridization (smFISH). 

In a smFISH experiment the number of TREG puncta can be used to infer the total
RNA expression of the cell. The shape defined by the TREG puncta would help 
define the shape and size of a cell.

The motivation of this work is to collect data via smFISH in to help build better 
deconvolution algorithms. But there are many application for TREGs in experimental 
design!

### What makes a gene a good TREG?  
1. The gene must have non-zero expression in most cells across different tissue 
and cell types. This is facilitated with the functions `get_prop_zero` and 
`filter_prop_zero`

2. A TREG should also be expressed at a constant level in respect to other genes
across different cell types or have *high rank invariance*. Genes in a dataset 
can be evaluated for their invariance with the `rank_invariance_express`. Genes
with high invatiance are good candidate TREGs.

3. Be measurable as a continuous metric in the experimental assay, for example
have a dynamic range of puncta when observed in RANscope. This will need to be
considered for the canidate TREGs, and may need to be validated experimentally. 


![RI Flow](man/figures/Figure1.png)

## Installation instructions

Get the latest stable `R` release from [CRAN](http://cran.r-project.org/). Then install `TREG` using from [Bioconductor](http://bioconductor.org/) the following code:

```{r 'install', eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("TREG")
```

And the development version from [GitHub](https://github.com/LieberInstitute/TREG) with:

```{r 'install_dev', eval = FALSE}
BiocManager::install("LieberInstitute/TREG")
```
## Example

### Proption Zero Filtering  
A Good TREG candidate gene will be expressed in almost every cell. So the list of
genes should be filtered by proportion of zeros across groups of cells.

```{r calc_prop_zero, eval = requireNamespace('TREG')}
library("TREG")
## Calclualte Proportion Zero
prop_zero <- get_prop_zero(sce = sce_zero_test, group_col = "cellType")
(canidate_genes <- filter_prop_zero(prop_zero))
```
### Evaluate RI for Example Data

```{r run_RI, eval = requireNamespace('TREG')}

sce_filter <- sce_zero_test[canidate_genes,]
dim(sce_filter)
rank_invariance_express(sce_filter)

```


## Citation

Below is the citation output from using `citation('TREG')` in R. Please
run this yourself to check for any updates on how to cite __TREG__.

```{r 'citation', eval = requireNamespace('TREG')}
print(citation("TREG"), bibtex = TRUE)
```

Please note that the `TREG` was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.

## Code of Conduct

Please note that the `TREG` project is released with a [Contributor Code of Conduct](http://bioconductor.org/about/code-of-conduct/). By contributing to this project, you agree to abide by its terms.

## Development tools

* Continuous code testing is possible thanks to [GitHub actions](https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/)  through `r BiocStyle::CRANpkg('usethis')`, `r BiocStyle::CRANpkg('remotes')`, and `r BiocStyle::CRANpkg('rcmdcheck')` customized to use [Bioconductor's docker containers](https://www.bioconductor.org/help/docker/) and `r BiocStyle::Biocpkg('BiocCheck')`.
* Code coverage assessment is possible thanks to [codecov](https://codecov.io/gh) and `r BiocStyle::CRANpkg('covr')`.
* The [documentation website](http://LieberInstitute.github.io/TREG) is automatically updated thanks to `r BiocStyle::CRANpkg('pkgdown')`.
* The code is styled automatically thanks to `r BiocStyle::CRANpkg('styler')`.
* The documentation is formatted thanks to `r BiocStyle::CRANpkg('devtools')` and `r BiocStyle::CRANpkg('roxygen2')`.

For more details, check the `dev` directory.

This package was developed using `r BiocStyle::Biocpkg('biocthis')`.


