<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>How to Find TREGs • TREG</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="How to Find TREGs">
<meta property="og:description" content="TREG">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">TREG</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/how_to_find_tregs.html">How to Find TREGs</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/TREG/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>How to Find TREGs</h1>
                        <h4 data-toc-skip class="author">Louise A.
Huuki-Myers</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lahuuki@gmail.com" class="email">lahuuki@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">21 March 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/TREG/blob/HEAD/vignettes/how_to_find_tregs.Rmd" class="external-link"><code>vignettes/how_to_find_tregs.Rmd</code></a></small>
      <div class="hidden name"><code>how_to_find_tregs.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-treg">Install <code>TREG</code><a class="anchor" aria-label="anchor" href="#install-treg"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em> is a
<code>R</code> package available via Bioconductor. <code>R</code> can be
installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"TREG"</span><span class="op">)</span>

<span class="co">## Check that you have a valid Bioconductor installation</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em>
(Huuki-Myers and Collado-Torres, 2022) is based on many other packages
and in particular in those that have implemented the infrastructure
needed for dealing with single cell RNA sequencing data, visualization
functions, and interactive data exploration. That is, packages like
<em><a href="https://bioconductor.org/packages/3.15/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
that allow you to store the data.</p>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But <code>R</code> and
<code>Bioconductor</code> have a steep learning curve so it is critical
to learn where to ask for help. The blog post quoted above mentions some
but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
as the main resource for getting help regarding Bioconductor. Other
alternatives are available such as creating GitHub issues and tweeting.
However, please note that if you want to receive help you should adhere
to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
<div class="section level3">
<h3 id="citing-treg">Citing <code>TREG</code><a class="anchor" aria-label="anchor" href="#citing-treg"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em> will be
useful for your research. Please use the following information to cite
the package and the research article describing the data provided by
<em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em>.
Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Citation info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"TREG"</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Huuki-Myers LA, Collado-Torres L (2022). _TREG: a R/Bioconductor</span>
<span class="co">#&gt; package to identify Total RNA Expression Genes_.</span>
<span class="co">#&gt; \Sexpr[results=rd]{tools:::Rd_expr_doi("10.18129/B9.bioc.TREG")},</span>
<span class="co">#&gt; https://github.com/LieberInstitute/TREG/TREG - R package version</span>
<span class="co">#&gt; 0.99.0, &lt;URL: http://www.bioconductor.org/packages/TREG&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Huuki-Myers LA, Montgomery KD, Kwon SH, Page SC, Hicks SC, Maynard KR,</span>
<span class="co">#&gt; Collado-Torres L (2022). "Data Driven Identification of Total RNA</span>
<span class="co">#&gt; Expression Genes "TREGs" for estimation of RNA abundance in</span>
<span class="co">#&gt; heterogeneous cell types." _bioRxiv_.</span>
<span class="co">#&gt; \Sexpr[results=rd]{tools:::Rd_expr_doi("10.1101/TODO")}, &lt;URL:</span>
<span class="co">#&gt; https://doi.org/10.1101/TODO&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span>
<span class="co">#&gt; bibtex=TRUE)', 'toBibtex(.)', or set</span>
<span class="co">#&gt; 'options(citation.bibtex.max=999)'.</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em>
(Huuki-Myers and Collado-Torres, 2022) package was developed for
identifying candidate <strong>Total RNA Expression Genes
(TREGs)</strong> for estimating RNA abundance for individual cells in an
snFISH experiment by researchers at the Lieber Institute for Brain
Development (LIBD) (Huuki-Myers, Montgomery, Kwon, Page, Hicks, Maynard,
and Collado-Torres, 2022).</p>
<p>In this vignette we’ll showcase how you can use the R functions
provided by <em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em>
(Huuki-Myers and Collado-Torres, 2022) with the snRNA-seq dataset that
was recently published by our LIBD collaborators (Tran, Maynard,
Spangler, Huuki, Montgomery, Sadashivaiah, Tippani, Barry, Hancock,
Hicks, Kleinman, Hyde, Collado-Torres, Jaffe, and Martinowich,
2021).</p>
<p>To get started, please load the <em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em>
package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/TREG" class="external-link">"TREG"</a></span><span class="op">)</span></code></pre></div>
<p>The goal of <code>TREG</code> is to help find find candidate
<strong>Total RNA Expression Genes (TREGs)</strong> in single nucleus
(or single cell) RNA-seq data.</p>
<div class="section level3">
<h3 id="why-are-tregs-useful">Why are TREGs useful?<a class="anchor" aria-label="anchor" href="#why-are-tregs-useful"></a>
</h3>
<p>The expression of a TREG is proportional to the the overall RNA
expression in a cell. This relationship can be used to estimate total
RNA content in cells in assays where only a few genes can be measured,
such as single-molecule fluorescent in situ hybridization (smFISH).</p>
<p>In a smFISH experiment the number of TREG puncta can be used to infer
the total RNA expression of the cell.</p>
<p>The motivation of this work is to collect data via smFISH in to help
build better deconvolution algorithms. But may be many other application
for TREGs in experimental design!</p>
<p align="center">
</p>
<div class="figure">
<img src="http://research.libd.org/TREG/reference/figures/TREG_cartoon.png" style="width:50.0%" alt=""><p class="caption">The Expression of a TREG can inform total RNA content
of a cell</p>
</div>

</div>
<div class="section level3">
<h3 id="what-makes-a-gene-a-good-treg">What makes a gene a good TREG?<a class="anchor" aria-label="anchor" href="#what-makes-a-gene-a-good-treg"></a>
</h3>
<ol style="list-style-type: decimal">
<li><p>The gene must have <strong>non-zero expression in most
cells</strong> across different tissue and cell types.</p></li>
<li><p>A TREG should also be expressed at a constant level in respect to
other genes across different cell types or have <strong>high rank
invariance</strong>.</p></li>
<li><p>Be <strong>measurable as a continuous metric</strong> in the
experimental assay, for example have a dynamic range of puncta when
observed in RNAscope. This will need to be considered for the candidate
TREGs, and may need to be validated experimentally.</p></li>
</ol>
<p align="center">
</p>
<div class="figure">
<img src="http://research.libd.org/TREG/reference/figures/fig1_rank_violin_demo.png" style="width:30.0%" alt=""><p class="caption">Distribution of ranks of a gene of High and Low
Invariance</p>
</div>

</div>
<div class="section level3">
<h3 id="how-to-find-candidate-tregs-with-treg">How to find candidate TREGs with <code>TREG</code><a class="anchor" aria-label="anchor" href="#how-to-find-candidate-tregs-with-treg"></a>
</h3>
<p align="center">
</p>
<div class="figure">
<img src="http://research.libd.org/TREG/reference/figures/RI_flow.png" style="width:100.0%" alt=""><p class="caption">Overview of the Rank Invariance Process</p>
</div>

<ol style="list-style-type: decimal">
<li><p><strong>Filter for low Proportion Zero genes snRNA-seq
dataset:</strong> This is facilitated with the functions
<code><a href="../reference/get_prop_zero.html">get_prop_zero()</a></code> and <code><a href="../reference/filter_prop_zero.html">filter_prop_zero()</a></code>
(Equation @ref(eq:propZero)). snRNA-seq data is notoriously sparse,
these functions enrich for genes with more universal
expression.</p></li>
<li><p><strong>Evaluate genes for Rank Invariance</strong> The nuclei
are grouped only by cell type. Within each cell type, the mean
expression for each gene is ranked, the result is a vector (length is
the number of genes), using the function <code>rank_gene()</code>. Then
the expression of each gene is ranked for each nucleus,the result is a
matrix (the number of nuclei x number of genes), using the function
<code><a href="../reference/rank_cells.html">rank_cells()</a></code>.Then the absolute difference between the rank
of each nucleus and the mean expression is found, from here the mean of
the differences for each gene is calculated, then ranked. These steps
are repeated for each group, the result is a matrix of ranks, (number of
cell types x number of genes). From here the sum of the ranks for each
gene are reversed ranked, so there is one final value for each gene, the
“rank invariance” The genes with the highest rank-invariance are
considered good candidates as TREGs. This is calculated with
<code><a href="../reference/rank_invariance_express.html">rank_invariance_express()</a></code>. <strong>This full process is
implemented by:
<code><a href="../reference/rank_invariance_express.html">rank_invariance_express()</a></code>.</strong></p></li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="example-treg-application">Example TREG Application<a class="anchor" aria-label="anchor" href="#example-treg-application"></a>
</h2>
<p>In this example we will apply our data driven process for TREG
discovery to a snRNA-seq dataset. This process has three main
steps:<br>
1. Data prep<br>
2. Gene filtering: dropping genes with low expression and high
Proportion Zero (Equation @ref(eq:propZero))<br>
3. Rank Invariance Calculation</p>
<div class="section level3">
<h3 id="load-packages">Load Packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"SingleCellExperiment"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"pheatmap"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://tidyr.tidyverse.org" class="external-link">"tidyr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://tibble.tidyverse.org/" class="external-link">"tibble"</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="download-and-prep-data">Download and Prep Data<a class="anchor" aria-label="anchor" href="#download-and-prep-data"></a>
</h3>
<p>Here we download a public single nucleus RNA-seq (snRNA-seq) data
from (Tran, Maynard, Spangler, et al., 2021) that we’ll use as our
example. This data can be accessed on <a href="https://github.com/LieberInstitute/10xPilot_snRNAseq-human#processed-data" class="external-link">github</a>.
This data is from postmortem human brain in the dorsolateral prefrontal
cortex (DLPFC) region, and contains gene expression data for 11k
nuclei.</p>
<p>We will use <code>BiocFileCache()</code> to cache this data. It is
stored as a <code>SingleCellExperiment</code> object named
<code>sce.dlpfc.tran</code>, and takes 1.01 GB of RAM memory to
load.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Download and save a local cache of the data available at:</span>
<span class="co"># https://github.com/LieberInstitute/10xPilot_snRNAseq-human#processed-data</span>
<span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span>
    <span class="st">"https://libd-snrnaseq-pilot.s3.us-east-2.amazonaws.com/"</span>,
    <span class="st">"SCE_DLPFC-n3_tran-etal.rda"</span>
<span class="op">)</span>
<span class="va">local_data</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcrpath</a></span><span class="op">(</span><span class="va">url</span>, x <span class="op">=</span> <span class="va">bfc</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">local_data</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Loading objects:</span>
<span class="co">#&gt;   sce.dlpfc.tran</span></code></pre></div>
<table class="table">
<caption>Cell type names and corresponding acronyms used in this
dataset</caption>
<thead><tr class="header">
<th align="left">Cell Type</th>
<th align="left">Acronym</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Astrocyte</td>
<td align="left">Astro</td>
</tr>
<tr class="even">
<td align="left">Excitatory Neurons</td>
<td align="left">Excit</td>
</tr>
<tr class="odd">
<td align="left">Microglia</td>
<td align="left">Micro</td>
</tr>
<tr class="even">
<td align="left">Oligodendrocytes</td>
<td align="left">Oligo</td>
</tr>
<tr class="odd">
<td align="left">Oligodendrocyte Progenitor Cells</td>
<td align="left">OPC</td>
</tr>
<tr class="even">
<td align="left">Inhibitory Neurons</td>
<td align="left">Inhib</td>
</tr>
</tbody>
</table>
<p>Human brain tissue consists of many types of cells, for the porpose
of this demo, we will focus on the six major cell types listed in Table
@ref(tab:acronyms).</p>
<div class="section level4">
<h4 id="filter-and-refine-to-cell-types-of-interest">Filter and Refine to Cell Types of Interest<a class="anchor" aria-label="anchor" href="#filter-and-refine-to-cell-types-of-interest"></a>
</h4>
<p>First we will combine all of the Excit, Inhib subtypes, as it is a
finer resolution than we want to examine, and combine rare subtypes in
to one group. If there are too few cells in a group there may not be
enough data to get good results. This new cell type classification is
stored in the <code>colData</code> as <code>cellType.broad</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Explore the dimensions and cell type annotations</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span>
<span class="co">#&gt; [1] 33538 11202</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">$</span><span class="va">cellType</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Astro    Excit_A    Excit_B    Excit_C    Excit_D    Excit_E    Excit_F </span>
<span class="co">#&gt;        782        529        773        524        132        187        243 </span>
<span class="co">#&gt;    Inhib_A    Inhib_B    Inhib_C    Inhib_D    Inhib_E    Inhib_F Macrophage </span>
<span class="co">#&gt;        333        454        365        413          7          8         10 </span>
<span class="co">#&gt;      Micro      Mural      Oligo        OPC      Tcell </span>
<span class="co">#&gt;        388         18       5455        572          9</span>

<span class="co">## Use a lower resolution of cell type annotation</span>
<span class="va">sce.dlpfc.tran</span><span class="op">$</span><span class="va">cellType.broad</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_[A-Z]$"</span>, <span class="st">""</span>, <span class="va">sce.dlpfc.tran</span><span class="op">$</span><span class="va">cellType</span><span class="op">)</span>
<span class="op">(</span><span class="va">cell_type_tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">$</span><span class="va">cellType.broad</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Astro      Excit      Inhib Macrophage      Micro      Mural      Oligo </span>
<span class="co">#&gt;        782       2388       1580         10        388         18       5455 </span>
<span class="co">#&gt;        OPC      Tcell </span>
<span class="co">#&gt;        572          9</span></code></pre></div>
<p>Next,we will drop any groups with &lt; 50 cells after merging
subtypes. This excludes any very rare cell types. Now we are working
with the six broad cell type we are interested in.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Find cell types with &lt; 50 cells</span>
<span class="op">(</span><span class="va">ct_drop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cell_type_tab</span><span class="op">)</span><span class="op">[</span><span class="va">cell_type_tab</span> <span class="op">&lt;</span> <span class="fl">50</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; [1] "Macrophage" "Mural"      "Tcell"</span>

<span class="co">## Filter columns of sce object</span>
<span class="va">sce.dlpfc.tran</span> <span class="op">&lt;-</span> <span class="va">sce.dlpfc.tran</span><span class="op">[</span>, <span class="op">!</span><span class="va">sce.dlpfc.tran</span><span class="op">$</span><span class="va">cellType.broad</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">ct_drop</span><span class="op">]</span>

<span class="co">## Check new cell type bread down and dimension</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">$</span><span class="va">cellType.broad</span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Astro Excit Inhib Micro Oligo   OPC </span>
<span class="co">#&gt;   782  2388  1580   388  5455   572</span>

<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span>
<span class="co">#&gt; [1] 33538 11165</span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="filter-genes">Filter Genes<a class="anchor" aria-label="anchor" href="#filter-genes"></a>
</h3>
<p>Single Nucleus data is often very sparse (lots of zeros in the count
data), this dataset is 88% sparse. We can illustrate this in the heat
map of the first 1k genes and 500 cells. The heatmap is mostly blue,
indicating low values (Figure @ref(fig:examineSparsity)).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## this data is 88% sparse</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.8839022</span>

<span class="co">## lets make a heatmap of the first 1k genes and 500 cells</span>
<span class="va">count_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">500</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">count_test</span>,
    cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,
    cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,
    show_rownames <span class="op">=</span> <span class="cn">FALSE</span>,
    show_colnames <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<div class="figure">
<img src="how_to_find_tregs_files/figure-html/examineSparsity-1.png" alt="Heatmap of the snRNA-seq counts. Illustrates sparseness of unfiltered data." width="700"><p class="caption">
Heatmap of the snRNA-seq counts. Illustrates sparseness of unfiltered
data.
</p>
</div>
<div class="section level4">
<h4 id="filter-to-top-50-expression">Filter to Top 50% Expression<a class="anchor" aria-label="anchor" href="#filter-to-top-50-expression"></a>
</h4>
<p>Determine the median expression of genes over all rows, drop all the
genes that are below this limit.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">row_means</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">)</span>
<span class="op">(</span><span class="va">median_row_means</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">row_means</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.01405562</span>

<span class="va">sce.dlpfc.tran</span> <span class="op">&lt;-</span> <span class="va">sce.dlpfc.tran</span><span class="op">[</span><span class="va">row_means</span> <span class="op">&gt;</span> <span class="va">median_row_means</span>, <span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span>
<span class="co">#&gt; [1] 16769 11165</span></code></pre></div>
<p>After this filter lets check sparsity and make a heatmap of the first
1k genes and 500 cells. We are seeing more non-blue (Figure
@ref(fig:top50FilterHeatmap))!</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## this data down to 77% sparse</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.7713423</span>

<span class="co">## replot heatmap</span>
<span class="va">count_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">500</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu">pheatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">count_test</span>,
    cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,
    cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,
    show_rownames <span class="op">=</span> <span class="cn">FALSE</span>,
    show_colnames <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<div class="figure">
<img src="how_to_find_tregs_files/figure-html/top50FilterHeatmap-1.png" alt="Heatmap of the snRNA-seq counts. With top 50% filtering the data becomes less sparse." width="700"><p class="caption">
Heatmap of the snRNA-seq counts. With top 50% filtering the data becomes
less sparse.
</p>
</div>
</div>
<div class="section level4">
<h4 id="calculate-proportion-zero-and-pick-cutoff">Calculate Proportion Zero and Pick Cutoff<a class="anchor" aria-label="anchor" href="#calculate-proportion-zero-and-pick-cutoff"></a>
</h4>
<p>For each group (let’s use <code>cellType.broad</code>) get the
Proportion Zero for each gene, where Proportion Zero is defined in
Equation @ref(eq:propZero) where <span class="math inline">\(c_{i,j,k,z}\)</span> is the number of snRNA-seq
counts for cell/nucleus <span class="math inline">\(z\)</span> for gene
<span class="math inline">\(i\)</span>, cell type <span class="math inline">\(j\)</span>, brain region <span class="math inline">\(k\)</span>, and <span class="math inline">\(n_{j,
k}\)</span> is the number of cells/nuclei for cell type <span class="math inline">\(j\)</span> and brain region <span class="math inline">\(k\)</span>.</p>
<p><span class="math display">\[\begin{equation}
    PZ_{i,j,k} = \sum_{z=1}^{n_{j,k}} I(c_{i,j,k} &gt; 0) / n_{j,k}
  (\#eq:propZero)
\end{equation}\]</span></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get prop zero for each gene for each cell type</span>
<span class="va">prop_zeros</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_prop_zero.html">get_prop_zero</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span>, group_col <span class="op">=</span> <span class="st">"cellType.broad"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prop_zeros</span><span class="op">)</span>
<span class="co">#&gt;                Astro     Excit     Inhib     Micro     Oligo       OPC</span>
<span class="co">#&gt; AL627309.1 0.9667519 0.7784757 0.8487342 0.9536082 0.9552704 0.9685315</span>
<span class="co">#&gt; AL669831.5 0.8836317 0.2361809 0.4879747 0.8608247 0.8445463 0.7657343</span>
<span class="co">#&gt; LINC00115  0.9948849 0.9493300 0.9759494 0.9845361 0.9891842 0.9930070</span>
<span class="co">#&gt; NOC2L      0.9347826 0.6168342 0.7556962 0.9536082 0.9365720 0.8811189</span>
<span class="co">#&gt; KLHL17     0.9859335 0.8986600 0.9360759 0.9922680 0.9963336 0.9877622</span>
<span class="co">#&gt; HES4       0.8695652 0.7357621 0.6892405 0.9974227 0.9906508 0.9737762</span></code></pre></div>
<p>To determine a good cutoff for filtering lets examine the
distribution of these Proportion Zeros by group.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Pivot data longer for plotting</span>
<span class="va">prop_zero_long</span> <span class="op">&lt;-</span> <span class="va">prop_zeros</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"Gene"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="op">!</span><span class="va">Gene</span>, names_to <span class="op">=</span> <span class="st">"Group"</span>, values_to <span class="op">=</span> <span class="st">"prop_zero"</span><span class="op">)</span>

<span class="co"># Plot histograms</span>
<span class="op">(</span><span class="va">prop_zero_histogram</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>
    data <span class="op">=</span> <span class="va">prop_zero_long</span>,
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">prop_zero</span>, fill <span class="op">=</span> <span class="va">Group</span><span class="op">)</span>
<span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>binwidth <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Group</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="how_to_find_tregs_files/figure-html/propZeroDistribution-1.png" alt="Distribution of Proption Zero across cell types and regions." width="700"><p class="caption">
Distribution of Proption Zero across cell types and regions.
</p>
</div>
<p>Looks like around 0.9 the densities peak, we’ll set that as the
cutoff (Figure @ref(fig:propZeroDistribution)).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Specify a cutoff, here we use 0.9</span>
<span class="va">propZero_limit</span> <span class="op">&lt;-</span> <span class="fl">0.9</span>

<span class="co">## Add a vertical red dashed line where the cutoff is located</span>
<span class="va">prop_zero_histogram</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="va">propZero_limit</span>, color <span class="op">=</span> <span class="st">"red"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="how_to_find_tregs_files/figure-html/pickCutoff-1.png" alt="Show Proportion Zero Cutoff on Distributions." width="700"><p class="caption">
Show Proportion Zero Cutoff on Distributions.
</p>
</div>
<p>The chosen cutoff excludes the peak Proportion Zeros from all groups
(Figure @ref(fig:pickCutoff)).</p>
</div>
<div class="section level4">
<h4 id="filter-by-the-max-proportion-zero">Filter by the Max Proportion Zero<a class="anchor" aria-label="anchor" href="#filter-by-the-max-proportion-zero"></a>
</h4>
<p>Use the cutoff to filter the remaining genes. Only 4k or ~11% of
genes pass with this cutoff. Filter the SCE object to this set of
genes.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## get a list of filtered genes</span>
<span class="va">filtered_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_prop_zero.html">filter_prop_zero</a></span><span class="op">(</span><span class="va">prop_zeros</span>, cutoff <span class="op">=</span> <span class="va">propZero_limit</span><span class="op">)</span>

<span class="co">## How many genes pass the filter?</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">filtered_genes</span><span class="op">)</span>
<span class="co">#&gt; [1] 4005</span>

<span class="co">## What % of genes is this</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">filtered_genes</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.2388336</span>

<span class="co">## Filter the sce object</span>
<span class="va">sce.dlpfc.tran</span> <span class="op">&lt;-</span> <span class="va">sce.dlpfc.tran</span><span class="op">[</span><span class="va">filtered_genes</span>, <span class="op">]</span></code></pre></div>
<p>One last check of the sparsity, more non-blue means more non-zero
values for the Rank Invariance calculation, which prevents rank ties
(Figure @ref(fig:propZeroFilterHeatmap)).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## this data down to 50% sparse</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">$</span><span class="va">counts</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.5010587</span>

<span class="co">## re-plot heatmap</span>
<span class="va">count_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span><span class="op">)</span><span class="op">$</span><span class="va">logcounts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">1000</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">500</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="fu">pheatmap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">count_test</span>,
    cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,
    cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>,
    show_rownames <span class="op">=</span> <span class="cn">FALSE</span>,
    show_colnames <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span></code></pre></div>
<div class="figure">
<img src="how_to_find_tregs_files/figure-html/propZeroFilterHeatmap-1.png" alt="Heatmap of the snRNA-seq counts after Proption Zero filtering." width="700"><p class="caption">
Heatmap of the snRNA-seq counts after Proption Zero filtering.
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="run-rank-invariance">Run Rank Invariance<a class="anchor" aria-label="anchor" href="#run-rank-invariance"></a>
</h3>
<p>To get the Rank Invariance (RI), the rank of the genes across the
cells in a group, and between groups is considered. One way to calculate
RI is to find the group rank values, and cell rank values separately,
then combine them as shown below. The genes with the top RI values are
the best candidate TREGs.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Get the rank of the gene in each group</span>
<span class="va">group_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rank_group.html">rank_group</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span>, group_col <span class="op">=</span> <span class="st">"cellType.broad"</span><span class="op">)</span>

<span class="co">## Get the rank of the gene for each cell</span>
<span class="va">cell_rank</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rank_cells.html">rank_cells</a></span><span class="op">(</span><span class="va">sce.dlpfc.tran</span>, group_col <span class="op">=</span> <span class="st">"cellType.broad"</span><span class="op">)</span>

<span class="co">## Use both rankings to calculate rank_invariance()</span>
<span class="va">rank_invar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rank_invariance.html">rank_invariance</a></span><span class="op">(</span><span class="va">group_rank</span>, <span class="va">cell_rank</span><span class="op">)</span>

<span class="co">## The top 5 Canidate TREGs:</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">rank_invar</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;  MALAT1  CTNND2     FTX    RERE FAM155A    GNAQ </span>
<span class="co">#&gt;    4005    4004    4003    4002    4001    4000</span></code></pre></div>
<p>The <code><a href="../reference/rank_invariance_express.html">rank_invariance_express()</a></code> function combines these
three steps into one function, and achieves the same results.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## rank_invariance_express() runs the previous functions for you</span>
<span class="va">rank_invar2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rank_invariance_express.html">rank_invariance_express</a></span><span class="op">(</span>
    <span class="va">sce.dlpfc.tran</span>,
    group_col <span class="op">=</span> <span class="st">"cellType.broad"</span>
<span class="op">)</span>
<span class="co">## Again the top 5 Canidate TREGs:</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">rank_invar2</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;  MALAT1  CTNND2     FTX    RERE FAM155A    GNAQ </span>
<span class="co">#&gt;    4005    4004    4003    4002    4001    4000</span>

<span class="co">## Check computationally that the results are identical</span>
<span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html" class="external-link">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">rank_invar</span>, <span class="va">rank_invar2</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>We have identified top candidate TREG genes from this dataset, by
applying Proportion Zero filtering and calculating the Rank Invariance
using the <code>TREG</code> package. This provides a list of candidate
genes that can be useful for estimating total RNA expression in assays
such as smFISH.</p>
<p>However, we are unable to assess other important qualities of these
genes that ensure they are experimentally compatible with the chosen
assay. For example in with smFISH with RNAscope such as its important
that a TREG be expressed at a level that individual puncta can be
accurately counted, and have a dynamic range of puncta. During
experimental validation we found that <em>MALAT1</em> was too highly
expresses in the human DLPFC to segment individual puncta, and ruled it
out as an experimentally useful TREG (Huuki-Myers, Montgomery, Kwon, et
al., 2022).</p>
<p>Therefore, we recommend that TREGs be evaluated in the assay or
analysis of choice you perform a validation experiment with a pilot
sample before implementing experiments using using it on rare and
valuable samples.</p>
<p>If you are designing a sc/snRNA-seq study to use as a reference for
deconvolution of bulk RNA-seq, we recommend that you generate
spatially-adjacent dissections in order to use them for RNAscope
experiments. By doing so, you could identify cell types in your
sc/snRNA-seq data, then identify candidate TREGs based on those cell
types, and use these candidate TREGs in your spatially-adjacent
dissections to quantify size and total RNA amounts for the cell types of
interest. Furthermore, the RNAscope data alone can be used as a gold
standard reference for cell fractions.</p>
<p>TREGs could be useful for other research purposes and other contexts
than the ones we envisioned.</p>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.15/TREG" class="external-link">TREG</a></em> package
(Huuki-Myers and Collado-Torres, 2022) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/BiocFileCache" class="external-link">BiocFileCache</a></em>
(Shepherd and Morgan, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=dplyr" class="external-link">dplyr</a></em> (Wickham,
François, Henry, and Müller, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em>
(Wickham, 2016)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=Matrix" class="external-link">Matrix</a></em> (Bates
and Maechler, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=pheatmap" class="external-link">pheatmap</a></em>
(Kolde, 2019)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=purrr" class="external-link">purrr</a></em> (Henry
and Wickham, 2020)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rafalib" class="external-link">rafalib</a></em>
(Irizarry and Love, 2015)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, McPherson, Luraschi, Ushey, Atkins, Wickham, Cheng,
Chang, and Iannone, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
(Morgan, Obenchain, Hester, and Pagès, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=tibble" class="external-link">tibble</a></em> (Müller
and Wickham, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=tidyr" class="external-link">tidyr</a></em> (Wickham
and Girlich, 2022)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create the vignette</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"TREG.Rmd"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Extract the R code</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"TREG.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span class="co">#&gt; [1] "2022-03-21 17:45:05 UTC"</span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span class="co">#&gt; Time difference of 1.464 mins</span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span>
<span class="co">#&gt;  version  R Under development (unstable) (2022-03-14 r81896)</span>
<span class="co">#&gt;  os       Ubuntu 20.04.4 LTS</span>
<span class="co">#&gt;  system   x86_64, linux-gnu</span>
<span class="co">#&gt;  ui       X11</span>
<span class="co">#&gt;  language en</span>
<span class="co">#&gt;  collate  en_US.UTF-8</span>
<span class="co">#&gt;  ctype    en_US.UTF-8</span>
<span class="co">#&gt;  tz       UTC</span>
<span class="co">#&gt;  date     2022-03-21</span>
<span class="co">#&gt;  pandoc   2.17.1.1 @ /usr/local/bin/ (via rmarkdown)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version   </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span>
<span class="co">#&gt;  assertthat             0.2.1      <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  Biobase              * 2.55.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocFileCache          2.3.4      <span style="color: #555555;">2022-01-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocGenerics         * 0.41.2     <span style="color: #555555;">2021-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocManager            1.30.16    <span style="color: #555555;">2021-06-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  BiocStyle            * 2.23.1     <span style="color: #555555;">2021-12-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  bit                    4.0.4      <span style="color: #555555;">2020-08-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  bit64                  4.0.5      <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  bitops                 1.0-7      <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  blob                   1.2.2      <span style="color: #555555;">2021-07-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  bookdown               0.25       <span style="color: #555555;">2022-03-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  bslib                  0.3.1      <span style="color: #555555;">2021-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  cachem                 1.0.6      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  cli                    3.2.0      <span style="color: #555555;">2022-02-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  colorspace             2.0-3      <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  crayon                 1.5.0      <span style="color: #555555;">2022-02-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  curl                   4.3.2      <span style="color: #555555;">2021-06-23</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  DBI                    1.1.2      <span style="color: #555555;">2021-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  dbplyr                 2.1.1      <span style="color: #555555;">2021-04-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  DelayedArray           0.21.2     <span style="color: #555555;">2021-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  desc                   1.4.1      <span style="color: #555555;">2022-03-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  digest                 0.6.29     <span style="color: #555555;">2021-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  dplyr                * 1.0.8      <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  ellipsis               0.3.2      <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  evaluate               0.15       <span style="color: #555555;">2022-02-18</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  fansi                  1.0.2      <span style="color: #555555;">2022-01-14</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  farver                 2.1.0      <span style="color: #555555;">2021-02-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  fastmap                1.1.0      <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  filelock               1.0.2      <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  fs                     1.5.2      <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  generics               0.1.2      <span style="color: #555555;">2022-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  GenomeInfoDb         * 1.31.6     <span style="color: #555555;">2022-03-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  GenomeInfoDbData       1.2.7      <span style="color: #555555;">2022-03-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  GenomicRanges        * 1.47.6     <span style="color: #555555;">2022-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  ggplot2              * 3.3.5      <span style="color: #555555;">2021-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  glue                   1.6.2      <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  gtable                 0.3.0      <span style="color: #555555;">2019-03-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  highr                  0.9        <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  htmltools              0.5.2      <span style="color: #555555;">2021-08-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  httr                   1.4.2      <span style="color: #555555;">2020-07-20</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  IRanges              * 2.29.1     <span style="color: #555555;">2021-11-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  jquerylib              0.1.4      <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  jsonlite               1.8.0      <span style="color: #555555;">2022-02-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  knitr                  1.37       <span style="color: #555555;">2021-12-16</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  labeling               0.4.2      <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  lattice                0.20-45    <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  lifecycle              1.0.1      <span style="color: #555555;">2021-09-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  lubridate              1.8.0      <span style="color: #555555;">2021-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  magrittr               2.0.2      <span style="color: #555555;">2022-01-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  Matrix                 1.4-0      <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  MatrixGenerics       * 1.7.0      <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  matrixStats          * 0.61.0     <span style="color: #555555;">2021-09-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  memoise                2.0.1      <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  munsell                0.5.0      <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  pheatmap             * 1.0.12     <span style="color: #555555;">2019-01-04</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  pillar                 1.7.0      <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  pkgconfig              2.0.3      <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  pkgdown                <span style="color: #BB00BB; font-weight: bold;">2.0.2.9000</span> <span style="color: #555555;">2022-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">Github (r-lib/pkgdown@98d5a5c)</span></span>
<span class="co">#&gt;  plyr                   1.8.6      <span style="color: #555555;">2020-03-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  purrr                  0.3.4      <span style="color: #555555;">2020-04-17</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  R6                     2.5.1      <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  rafalib                1.0.0      <span style="color: #555555;">2015-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  ragg                   1.2.2      <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  rappdirs               0.3.3      <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  RColorBrewer           1.1-2      <span style="color: #555555;">2014-12-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  Rcpp                   1.0.8.3    <span style="color: #555555;">2022-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  RCurl                  1.98-1.6   <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  RefManageR           * 1.3.0      <span style="color: #555555;">2020-11-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  rlang                  1.0.2      <span style="color: #555555;">2022-03-04</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  rmarkdown              2.13       <span style="color: #555555;">2022-03-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  rprojroot              2.0.2      <span style="color: #555555;">2020-11-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  RSQLite                2.2.10     <span style="color: #555555;">2022-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  S4Vectors            * 0.33.11    <span style="color: #555555;">2022-03-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  sass                   0.4.0      <span style="color: #555555;">2021-05-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  scales                 1.1.1      <span style="color: #555555;">2020-05-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  sessioninfo          * 1.2.2      <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  SingleCellExperiment * 1.17.2     <span style="color: #555555;">2021-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  stringi                1.7.6      <span style="color: #555555;">2021-11-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  stringr                1.4.0      <span style="color: #555555;">2019-02-10</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  SummarizedExperiment * 1.25.3     <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  systemfonts            1.0.4      <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  textshaping            0.3.6      <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  tibble               * 3.1.6      <span style="color: #555555;">2021-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  tidyr                * 1.2.0      <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  tidyselect             1.1.2      <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  TREG                 * 0.99.0     <span style="color: #555555;">2022-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  utf8                   1.2.2      <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  vctrs                  0.3.8      <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  withr                  2.5.0      <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  xfun                   0.30       <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  xml2                   1.3.3      <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  XVector                0.35.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  yaml                   2.3.5      <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span>
<span class="co">#&gt;  zlibbioc               1.41.0     <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span>
<span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span>
<span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.15/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2021), <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2021) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, McPherson, et al., 2022) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2022rmarkdown"></a><a href="#cite-allaire2022rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, J. McPherson, et al. <em>rmarkdown: Dynamic
Documents for R</em>. R package version 2.13. 2022. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bates2021matrix"></a><a href="#cite-bates2021matrix">[2]</a><cite>
D. Bates and M. Maechler. <em>Matrix: Sparse and Dense Matrix Classes
and Methods</em>. R package version 1.4-0. 2021. URL:
<a href="https://CRAN.R-project.org/package=Matrix" class="external-link">https://CRAN.R-project.org/package=Matrix</a>.</cite>
</p>
<p>
<a id="bib-henry2020purrr"></a><a href="#cite-henry2020purrr">[3]</a><cite>
L. Henry and H. Wickham. <em>purrr: Functional Programming Tools</em>. R
package version 0.3.4. 2020. URL:
<a href="https://CRAN.R-project.org/package=purrr" class="external-link">https://CRAN.R-project.org/package=purrr</a>.</cite>
</p>
<p>
<a id="bib-huukimyers2022treg"></a><a href="#cite-huukimyers2022treg">[4]</a><cite>
L. A. Huuki-Myers and L. Collado-Torres. <em>TREG: a R/Bioconductor
package to identify Total RNA Expression Genes</em>. <a href="https://github.com/LieberInstitute/TREG/TREG" class="external-link uri">https://github.com/LieberInstitute/TREG/TREG</a> - R package
version 0.99.0. 2022. DOI:
<a href="https://doi.org/10.18129/B9.bioc.TREG" class="external-link">10.18129/B9.bioc.TREG</a>.
URL:
<a href="http://www.bioconductor.org/packages/TREG" class="external-link">http://www.bioconductor.org/packages/TREG</a>.</cite>
</p>
<p>
<a id="bib-huukimyers2022data"></a><a href="#cite-huukimyers2022data">[5]</a><cite>
L. A. Huuki-Myers, K. D. Montgomery, S. H. Kwon, et al. “Data Driven
Identification of Total RNA Expression Genes "TREGs" for estimation of
RNA abundance in heterogeneous cell types”. In: <em>bioRxiv</em> (2022).
DOI: <a href="https://doi.org/10.1101/TODO" class="external-link">10.1101/TODO</a>. URL:
<a href="https://doi.org/10.1101/TODO" class="external-link">https://doi.org/10.1101/TODO</a>.</cite>
</p>
<p>
<a id="bib-irizarry2015rafalib"></a><a href="#cite-irizarry2015rafalib">[6]</a><cite>
R. A. Irizarry and M. I. Love. <em>rafalib: Convenience Functions for
Routine Data Exploration</em>. R package version 1.0.0. 2015. URL:
<a href="https://CRAN.R-project.org/package=rafalib" class="external-link">https://CRAN.R-project.org/package=rafalib</a>.</cite>
</p>
<p>
<a id="bib-kolde2019pheatmap"></a><a href="#cite-kolde2019pheatmap">[7]</a><cite>
R. Kolde. <em>pheatmap: Pretty Heatmaps</em>. R package version 1.0.12.
2019. URL:
<a href="https://CRAN.R-project.org/package=pheatmap" class="external-link">https://CRAN.R-project.org/package=pheatmap</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[8]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-morgan2021summarizedexperiment"></a><a href="#cite-morgan2021summarizedexperiment">[9]</a><cite>
M. Morgan, V. Obenchain, J. Hester, et al. <em>SummarizedExperiment:
SummarizedExperiment container</em>. R package version 1.25.3. 2021.
URL:
<a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">https://bioconductor.org/packages/SummarizedExperiment</a>.</cite>
</p>
<p>
<a id="bib-mller2021tibble"></a><a href="#cite-mller2021tibble">[10]</a><cite>
K. Müller and H. Wickham. <em>tibble: Simple Data Frames</em>. R package
version 3.1.6. 2021. URL:
<a href="https://CRAN.R-project.org/package=tibble" class="external-link">https://CRAN.R-project.org/package=tibble</a>.</cite>
</p>
<p>
<a id="bib-ole2021biocstyle"></a><a href="#cite-ole2021biocstyle">[11]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.23.1. 2021. URL:
<a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2022language"></a><a href="#cite-2022language">[12]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2022. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-shepherd2022biocfilecache"></a><a href="#cite-shepherd2022biocfilecache">[13]</a><cite>
L. Shepherd and M. Morgan. <em>BiocFileCache: Manage Files Across
Sessions</em>. R package version 2.3.4. 2022.</cite>
</p>
<p>
<a id="bib-tran2021"></a><a href="#cite-tran2021">[14]</a><cite> M. N.
Tran, K. R. Maynard, A. Spangler, et al. “Single-nucleus transcriptome
analysis reveals cell-type-specific molecular signatures across reward
circuitry in the human brain”. In: <em>Neuron</em> (2021). DOI:
<a href="https://doi.org/10.1016/j.neuron.2021.09.001" class="external-link">10.1016/j.neuron.2021.09.001</a>.</cite>
</p>
<p>
<a id="bib-wickham2016ggplot2"></a><a href="#cite-wickham2016ggplot2">[15]</a><cite>
H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>.
Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. URL:
<a href="https://ggplot2.tidyverse.org" class="external-link">https://ggplot2.tidyverse.org</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[16]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[17]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2. 2021. URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-wickham2022dplyr"></a><a href="#cite-wickham2022dplyr">[18]</a><cite>
H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. R package version 1.0.8. 2022. URL:
<a href="https://CRAN.R-project.org/package=dplyr" class="external-link">https://CRAN.R-project.org/package=dplyr</a>.</cite>
</p>
<p>
<a id="bib-wickham2022tidyr"></a><a href="#cite-wickham2022tidyr">[19]</a><cite>
H. Wickham and M. Girlich. <em>tidyr: Tidy Messy Data</em>. R package
version 1.2.0. 2022. URL:
<a href="https://CRAN.R-project.org/package=tidyr" class="external-link">https://CRAN.R-project.org/package=tidyr</a>.</cite>
</p>
<p>
<a id="bib-xie2021knitr"></a><a href="#cite-xie2021knitr">[20]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.37. 2021. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Louise Huuki.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
